pkg/component/etcd/statefulset/statefulset.go

```diff
@@ -286,7 +286,8 @@ func immutableFieldUpdate(sts *appsv1.StatefulSet, val Values) bool {
 
 func clusterScaledUpToMultiNode(val *Values, sts *appsv1.StatefulSet) bool {
 	if sts != nil && sts.Spec.Replicas != nil {
-		return val.Replicas > 1 && *sts.Spec.Replicas == 1
+		return (val.Replicas > 1 && *sts.Spec.Replicas == 1) ||
```

Comment:
can you enhance the unit tests to include all cases?

Reply:
I don't think so there is an already existing unit tests for this.